python/ray/rllib/evaluation/sampler.py

```diff
@@ -290,6 +290,7 @@ def new_episode():
                 "env": base_env,
                 "policy": policies,
                 "episode": episode,
+                "info": info
```

Comment:
Where is info defined?

Reply:
We receive the info dict here in sampler.py:

unfiltered_obs, rewards, dones, infos, off_policy_actions = \
       base_env.poll()

It will now contain additional information that on_episode_start callback needs. Where I added the info as an input is where the callback function is executed. So I'm adding it there.

Reply:
if you mean line 303 (below this one), there the variable is infoS and this one is without the s. Is it working?

The change looks good to me. 

Just a note: it will be more clear and safe if the variables used by the nested function are already in the closure at the function definition point (or change it to be a function argument). In previous versions of python with static nested scopes/closures, this would not work I think.
I'm not sure if this is a convention in Python but for sure it's in formal functional languages.

Reply:
oh that was a typo. In my local container it was infos and worked fine. I fixed it here. Thanks for pointing that out.